test/intl402/Intl/getCanonicalLocales/overriden-arg-length.js

```diff
@@ -0,0 +1,115 @@
+// Copyright 2016 Mozilla Corporation. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
+
+/*---
+esid: sec-intl.getcanonicallocales
+description: Test Intl.getCanonicalLocales for step 5. 
+info: |
+  9.2.1 CanonicalizeLocaleList (locales)
+    5. Let len be ? ToLength(? Get(O, "length")).
+includes: [compareArray.js]
+features: [Symbol]
+---*/
+
+var locales = {
+  '0': 'en-US',
+};
+
+Object.defineProperty(locales, "length", {
+  get: function() { throw new Test262Error() }
+});
+
+assert.throws(Test262Error, function() {
+  Intl.getCanonicalLocales(locales);
+}, "should throw if locales.length throws");
+
+var locales = {
+  '0': 'en-US',
+  '1': 'pt-BR',
+};
+
+Object.defineProperty(locales, "length", {
+  get: function() { return "1" }
+});
+
+assert(compareArray(Intl.getCanonicalLocales(locales), ['en-US']),
+  "should return one element if locales.length is '1'");
+
+var locales = {
+  '0': 'en-US',
+  '1': 'pt-BR',
+};
+
+Object.defineProperty(locales, "length", {
+  get: function() { return 1.3 }
+});
+
+assert(compareArray(Intl.getCanonicalLocales(locales), ['en-US']),
+  "should return one element if locales.length is 1.3");
+
+var locales = {
+  '0': 'en-US',
+  '1': 'pt-BR',
+};
+
+Object.defineProperty(locales, "length", {
+  get: function() { return Symbol("1.8") }
+});
+
+assert.throws(TypeError, function() {
+  Intl.getCanonicalLocales(locales);
+}, "should throw if locales.length is a Symbol");
+
+var locales = {
+  '0': 'en-US',
+  '1': 'pt-BR',
+};
+
+Object.defineProperty(locales, "length", {
+  get: function() { return -Infinity }
+});
+
+assert(compareArray(Intl.getCanonicalLocales(locales), []),
+  "should return empty array if locales.length is -Infinity");
+
+var locales = {
+  '0': 'en-US',
+  '1': 'pt-BR',
+};
+
+Object.defineProperty(locales, "length", {
+  get: function() { return +Infinity }
+});
+
+assert(compareArray(Intl.getCanonicalLocales(locales), []),
+  "should return empty array if locales.length is +Infinity");
+
+var locales = {
+  '0': 'en-US',
+  '1': 'pt-BR',
+};
+
+Object.defineProperty(locales, "length", {
+  get: function() { return -5 }
+});
+
+assert(compareArray(Intl.getCanonicalLocales(locales), []),
+  "should return empty array if locales.length is a negative value");
+
+var locales = {
+  '0': 'en-US',
+  '1': 'pt-BR',
+};
+
+Object.defineProperty(locales, "length", {
+  get: function() { return Math.pow(2, 53); }
+});
+
+assert(compareArray(Intl.getCanonicalLocales(locales), []),
+  "should return empty array if locales.length is over 2**53-1");
```

Comment:
In a comment by @leobalter - https://github.com/tc39/test262/pull/642#issuecomment-220478113 he mentions:

```
step 5:

    check the Get(O, "length") operation, you may use an Array-like object patching a get accessor on the length property and returning a custom number.
    return custom abrupt completion from this length property (the same get accessor can throw a new Test262Error()
    values are coerced to an integer indexed length value on the ToLength (strings become numbers, float numbers become integers)
    throws a TypeError if length value is a Symbol
    length boundaries: -Infinity, negative values, +Infinity, values over 2**53-1, -0 becomes +0
```